src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -1738,16 +1740,29 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_AdvSimd_StoreSelectedScalar64x2:
         case NI_AdvSimd_StoreVector64x2:
+        case NI_AdvSimd_Arm64_StoreSelectedScalar128x2:
         case NI_AdvSimd_Arm64_StoreVector128x2:
         {
-            assert(sig->numArgs == 2);
             assert(retType == TYP_VOID);
 
             CORINFO_ARG_LIST_HANDLE arg1     = sig->args;
             CORINFO_ARG_LIST_HANDLE arg2     = info.compCompHnd->getArgNext(arg1);
             var_types               argType  = TYP_UNKNOWN;
             CORINFO_CLASS_HANDLE    argClass = NO_CLASS_HANDLE;
+            const bool isSingleStructStore = sig->numArgs == 3;
+
+            if (isSingleStructStore)
```

Comment:
These might need changes once we rename the APIs to `StoreSelectedScalar`.

Reply:
Done